### PR TITLE
Dockerfile: Add qemu for RISCV

### PR DESCRIPTION
**Pull request by martell**
I presume someone with access to dockerhub has to build this and merge it? :)

### CODE REVIEW

**Comment by nathanchance** [comment]
> I should probably make a check script that checks for both the binutils and QEMU binaries...

Done in #31. That should probably be merged first then RISCV can be added to the `scripts/check-binaries.sh` script.

**Comment by nathanchance** [comment]
Alright, this should be rebased on master so that the RISCV checks can be added to `scripts/check-binaries.sh`, then it should be good to go.

**Comment by martell** [comment]
Wow, I am very late to the party here.
I updated everything on top of the script improvements.

File: Dockerfile [hunk 1]
```diff
@@ -41,6 +41,8 @@ RUN apt-get update -qq && \
         qemu-system-arm \
         qemu-system-mips \
         qemu-system-ppc \
+        qemu-system-riscv32 \
+        qemu-system-riscv64 \
```
Comment (tpimh) [inline Dockerfile]: There are no such packages in Debian. The package that contains both `qemu-system-riscv32` and `qemu-system-riscv64` binaries is `qemu-system-misc`.

File: scripts/check-binaries.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ for BINARY in "${BINARIES[@]}"; do
     test_command "${BINARY}-${LLVM_VERSION}"
 done
 
-QEMU_SUFFIXES=( "arm" "aarch64" "i386" "mips" "mipsel" "ppc" "ppc64" "x86_64" )
+QEMU_SUFFIXES=( "arm" "aarch64" "i386" "mips" "mipsel" "ppc" "ppc64" "riscv32" "riscv64" "x86_64" )
```
Comment (tpimh) [inline scripts/check-binaries.sh]: Since `qemu-system-misc` also contains `qemu-system-s390x`, we can add it to this list as well.

Comment (martell) [reply scripts/check-binaries.sh]: done

File: Dockerfile (hunk 1 above)
Comment (martell) [reply Dockerfile]: Ahh, good catch, thank you

**Comment by nickdesaulniers** [comment]
Hmmm...that's not quite right; the debian package `opensbi` doesn't contain `opensbi-riscv64-virt-fw_jump.bin`. https://packages.debian.org/sid/all/opensbi/filelist

**Comment by nathanchance** [comment]
Do remember that we are on Ubuntu Focal now so make sure you are checking those packages. I am at work right now, I can look tonight when I am home.

**Comment by nickdesaulniers** [comment]
https://packages.ubuntu.com/focal/all/opensbi/filelist

looks like
/usr/lib/riscv64-linux-gnu/opensbi/qemu/virt/fw_jump.elf
gets installed, but qemu is searching for a file named
opensbi-riscv64-virt-fw_jump.bin

**Comment by nathanchance** [comment]
We can probably have some logic in boot-utils to find the right file. Does that file work if you tell QEMU to use it?

**Comment by tpimh** [comment]
Adding `-bios /usr/lib/riscv64-linux-gnu/opensbi/qemu/virt/fw_jump.elf` to qemu arguments list should be enough.

**Comment by nickdesaulniers** [comment]
Last night's RISCV boot tests [failed](https://travis-ci.com/github/ClangBuiltLinux/continuous-integration/jobs/351319505)

> qemu-system-riscv64 could not be found, please install it!

So we need this.  @martell , can you please rebase?

**Comment by nickdesaulniers** [comment]
and add `opensbi` to the list of installed packages?

**Comment by nickdesaulniers** [comment]
closing in favor of https://github.com/ClangBuiltLinux/dockerimage/pull/50

**Pull request closed without merging**